src/tiled/shapefilltool.cpp

```diff
@@ -219,14 +220,17 @@ void ShapeFillTool::updateFillOverlay()
     int left = std::min(mStartCorner.x(), mStartCorner.x() + width);
     int top = std::min(mStartCorner.y(), mStartCorner.y() + height);
 
+    QRect boundingRect(left, top, abs(width), abs(height));
+
     switch (mCurrentShape) {
     case Rect:
-        mFillRegion = QRegion(left, top, abs(width), abs(height),
-                              QRegion::Rectangle);
+        mFillRegion = QRegion(boundingRect, QRegion::Rectangle);
```

Comment:
Since rectangle is the default and the constructor taking a rectangle implicit, this could be shortened to:
```c++
mFillRegion = boundingRect;
```